### PR TITLE
PR2 Api Data interface implementations

### PR DESCRIPTION
**Pull request by danac-gs**
# Description
Split M2 Shipping and Tax endpoints

NOTE: Shipping and Tax split feature is divided into 4 PRs after it was finished. The split is done by dev workflow, position in the code base, execution logic and interference with the previous code (put in the last PR). However, the PRs are not independent of each other, that is visible in failed unit tests. The unit tests will pass when all 4 PRs get merged, as they do in the integral version at https://github.com/BoltApp/bolt-magento2/pull/641. That's why an intermediate branch is created, feature/shipping-and-tax-spli, instead of merging them directly into master.
 
Fixes: https://boltpay.atlassian.net/browse/M2P-51
https://github.com/BoltApp/bolt-magento2/pull/641

#changelog Api Data interface implementations

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

**Comment by danac-gs** [comment]
Renew

**Comment by danac-gs** [comment]
Renew

File: Test/Unit/Model/Api/Data/ShippingTaxTest.php [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 /**
  * Class ShippingTaxTest
- * @package Bolt\Boltpay\Test\Unit\Model\Api
+ * @package Bolt\Boltpay\Test\Unit\Model\Api\Data
```
Comment (matt-thomason) [inline Test/Unit/Model/Api/Data/ShippingTaxTest.php:24]: Good catch

File: Model/Api/Data/TaxData.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+{
+    /**
+     * @var TaxResultInterface
+     */
+    private $taxResult;
+
+    /**
+     * @var ShippingOptionInterface
+     */
+    private $shippingOption;
```
Comment (njfaries) [inline Model/Api/Data/TaxData.php:39]: Probably just revealing my own ignorance of M2's workings but if the shipping option has is doing it's own tax calculations, why does the tax data here need it?

Comment (danac-gs) [reply Model/Api/Data/TaxData.php:39]: @njfaries The first call, Shipping, returns the available shipping options and their prices but without the tax calculated. The Tax endpoint applies the selected shipping option to cart and then calculates all the taxes and returns subtotal tax and tax on shipping separately. It's the result model as defined in docs and expected on Bolt end. The shipping option has the same structure in both Shipping (output) and Tax (input and output), but the tax on shipping data is populated in Tax and left empty (zero) in Shipping.

**Pull request merged**